pkg/remote/aws/ec2_route_table_enumerator.go

```diff
@@ -0,0 +1,55 @@
+package aws
+
+import (
+	"github.com/cloudskiff/driftctl/pkg/remote/aws/repository"
+	remoteerror "github.com/cloudskiff/driftctl/pkg/remote/error"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+type EC2RouteTableEnumerator struct {
+	repository repository.EC2Repository
+	factory    resource.ResourceFactory
+}
+
+func NewEC2RouteTableEnumerator(repo repository.EC2Repository, factory resource.ResourceFactory) *EC2RouteTableEnumerator {
+	return &EC2RouteTableEnumerator{
+		repository: repo,
+		factory:    factory,
+	}
+}
+
+func (e *EC2RouteTableEnumerator) SupportedType() resource.ResourceType {
+	return aws.AwsRouteTableResourceType
+}
+
+func (e *EC2RouteTableEnumerator) Enumerate() ([]resource.Resource, error) {
+	routeTables, err := e.repository.ListAllRouteTables()
+	if err != nil {
+		return nil, remoteerror.NewResourceEnumerationError(err, string(e.SupportedType()))
+	}
+
+	var results []resource.Resource
+
+	for _, routeTable := range routeTables {
+		var isMain bool
```

Comment:
Maybe we can mutualize this logic in a method shared by the two route enumerators

Reply:
We can add a `isMainRouteTable` method in the `ec2_repository` file, wdyt ?

Reply:
I don't think that it is the repository responsibility to handle that